### ISSUE TITLE
ElectronCalibrator not calibrating data?

### ISSUE THREAD

**Issue by JamesSaxon** [UCATLAS/xAODAnaHelpers#283]
Hi folks -- 

I was trying to match a cut flow including electrons using 00-03-21, and I had a quick question.  I see the message:
>  Sample is Data! Do not apply any Electron Calibration... 

When I looked at the code, it seems like the ElectronCalibrator tool is not doing anything for data (just making a shallow copy).  

I think this is not the desired behavior; rather, as the tool notes at the top, one should have:

- scale corrections for DATA
- smearing corrections for MC
- (data VS. MC check is done by the CP tool internally) 

If I ignore m_isMC switch (line 342), I match the cut flow.  In earlier versions (00-03-08) there was not this "protection" against data, and I obtain the right value.

Is this just a bug, or is there some switch I need to change?

Thanks -- 

Jamie

**Comment by gfacini** [comment]
Looks like a bug to me...
Gabriel

**Comment by mmilesi** [comment]
Hi @JamesSaxon , @gfacini 

I guess you're right. I probably put that (wrong) protection for data because I checked this table:

https://twiki.cern.ch/twiki/bin/view/AtlasProtected/PhysicsAnalysisWorkBookRel20CPRec#Electrons

where it mentions the MC calibration only, and overlooked that note.

By the way, I have made a bunch of data/MC control plots, and indeed at the Zpeak for a dielectron selection I am seeing a slight shift of the Z peak in data towards lower mass. Most likely it will disappear with the correction. 

Thanks for helping me check all these!

I will fix this asap.

Marco
